<a href="https://colab.research.google.com/github/divyakedia/playstore-data-analysis/blob/main/(Divya_Self)Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

# 1. Importing the libraries necessary for project

In [167]:
#The first step is to import the libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline

# 2. Mounting google drive


In [168]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# 3. Creating path variable for the datasets

In [169]:
#creating data files path and reading the csv files
playstore_data_path = "/content/drive/MyDrive/(EDA)Capstone Project/Play Store Data (1).csv"
playstore_df = pd.read_csv(playstore_data_path)
users_review_path = "/content/drive/MyDrive/(EDA)Capstone Project/User Reviews.csv"
users_review_df = pd.read_csv(users_review_path)

# 4. Viewing the data of Python database

# 4.1 PlayStore Data

In [170]:
#Viewing the dataframe of playstore data
playstore_df.head(-1)

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10835                                           FR Forms             BUSINESS   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10835     NaN       0                9.6M          10+  Free     0   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   

      Content Rating                     Genres        Last Updated  \
0           Everyone               Art & Design     January 7, 2018   
1           Everyone  Art & Design;Pretend Play    January 15, 2018   
2           Everyone               Art & Design      August 1, 2018   
3               Teen               Art & Design        June 8, 2018   
4           Everyone    Art & Design;Creativity       June 20, 2018   
...              ...                        ...                 ...   
10835       Everyone                   Business  September 29, 2016   
10836       Everyone                  Education       July 25, 2017   
10837       Everyone                  Education        July 6, 2018   
10838       Everyone                    Medical    January 20, 2017   
10839     Mature 17+          Books & Reference    January 19, 2015   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10835               1.1.5          4.0 and up  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  

[10840 rows x 13 columns]

In [171]:
playstore_df.shape

(10841, 13)

In [172]:
# Print last 5 no. of rows
playstore_df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [173]:
# Printing basic information about the DataFrame
playstore_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


In [174]:
# To view available numeric column
playstore_df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

In [175]:
# To view all the columns of the available dataframe
playstore_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

# 4.2 User Review Data

In [176]:
users_review_df

App  \
0            10 Best Foods for You   
1            10 Best Foods for You   
2            10 Best Foods for You   
3            10 Best Foods for You   
4            10 Best Foods for You   
...                            ...   
64290  Houzz Interior Design Ideas   
64291  Houzz Interior Design Ideas   
64292  Houzz Interior Design Ideas   
64293  Houzz Interior Design Ideas   
64294  Houzz Interior Design Ideas   

                                       Translated_Review Sentiment  \
0      I like eat delicious food. That's I'm cooking ...  Positive   
1        This help eating healthy exercise regular basis  Positive   
2                                                    NaN       NaN   
3             Works great especially going grocery store  Positive   
4                                           Best idea us  Positive   
...                                                  ...       ...   
64290                                                NaN       NaN   
64291                                                NaN       NaN   
64292                                                NaN       NaN   
64293                                                NaN       NaN   
64294                                                NaN       NaN   

       Sentiment_Polarity  Sentiment_Subjectivity  
0                    1.00                0.533333  
1                    0.25                0.288462  
2                     NaN                     NaN  
3                    0.40                0.875000  
4                    1.00                0.300000  
...                   ...                     ...  
64290                 NaN                     NaN  
64291                 NaN                     NaN  
64292                 NaN                     NaN  
64293                 NaN                     NaN  
64294                 NaN                     NaN  

[64295 rows x 5 columns]

In [177]:
# To view available numeric column
users_review_df.describe()

Sentiment_Polarity  Sentiment_Subjectivity
count        37432.000000            37432.000000
mean             0.182146                0.492704
std              0.351301                0.259949
min             -1.000000                0.000000
25%              0.000000                0.357143
50%              0.150000                0.514286
75%              0.400000                0.650000
max              1.000000                1.000000

In [178]:
# Printing basic information about the DataFrame
users_review_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


# 5. Cleaning the data of Playstore Dataframe

By Studying the DataFrame, we see noticed that,


*  The Shape of the DataFrame is (10841, 13).
*    Column 'Reviews', 'Size', 'Installs' and 'Price' are in the type of 'object'.
*   Column 'Size' have the  strings representing size in 'M' as Megabytes, 'k' as kilobytes and also 'Varies with devices'.
*   Column 'Installs' have strings representing no. of installs with symbols such as ',' and '+'.
*   Column 'Price' have strings representing price with symbol '$'.

Thus, the data has to be cleaned for further process.





In [179]:
# Find the row with insufficiant data:
playstore_df[playstore_df['Type']!='Free'][playstore_df[playstore_df['Type']!='Free']['Price']=='0']


App Category  Rating Reviews                Size  \
9148  Command & Conquer: Rivals   FAMILY     NaN       0  Varies with device   

     Installs Type Price Content Rating    Genres   Last Updated  \
9148        0  NaN     0   Everyone 10+  Strategy  June 28, 2018   

             Current Ver         Android Ver  
9148  Varies with device  Varies with device

In [180]:
#Drop the row from the data frame:
playstore_df.drop(playstore_df[playstore_df['Type']!='Free'][playstore_df[playstore_df['Type']!='Free']['Price']=='0'].index, inplace=True)

In [181]:
#Find value mismatched row:
playstore_df[playstore_df['Genres']=='February 11, 2018']

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [182]:
#Find the index of the row which contains the carbage value:
playstore_df[playstore_df['Genres']=='February 11, 2018'].index

Int64Index([10472], dtype='int64')

In [183]:
#Drop the found carbage row from our dataframe:
playstore_df.drop(playstore_df[playstore_df['Genres']=='February 11, 2018'].index, inplace=True)

In [184]:
#making a fuction to clean the strings:
def clean_data(val):

  
  if 'M' in str(val):
    val = str(int(float(val.replace('M',''))*1000000))
  if 'k' in str(val):
    val = str(int(float(val.replace('k',''))*1000))
  if '+' in str(val):
    val = val.replace('+',' ')
  if ',' in str(val):
    val = val.replace(',','')
  if 'NaN' in str(val):
    val = '0'
  if '$' in str(val):
    val = val.replace('$','')
  else:
    pass
  return val

In [185]:
# To Clean the unwanted charactors and to convert the required column values into valid numeric type:

#Change the 'Reviews' column values to valid numeric values:
playstore_df['Reviews'] = pd.to_numeric(playstore_df['Reviews'])

#Change the 'Size' column values to valid numeric values:
playstore_df['Size'] = playstore_df['Size'].apply(lambda x: str(x).replace('Varies with device', 'NaN') if 'Varies with device' in str(x) else x)
playstore_df['Size'] = pd.to_numeric(playstore_df['Size'].map(lambda x: clean_data(x)))

#Change the 'Installs' column values to valid numeric values:
playstore_df['Installs'] = pd.to_numeric(playstore_df['Installs'].map(lambda x: clean_data(x)))

#Change the 'Price' column values to valid numeric values:
playstore_df['Price'] = pd.to_numeric (playstore_df['Price'].map(lambda x:clean_data(x)))
playstore_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10839 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10839 non-null  object 
 1   Category        10839 non-null  object 
 2   Rating          9366 non-null   float64
 3   Reviews         10839 non-null  int64  
 4   Size            10839 non-null  int64  
 5   Installs        10839 non-null  int64  
 6   Type            10839 non-null  object 
 7   Price           10839 non-null  float64
 8   Content Rating  10839 non-null  object 
 9   Genres          10839 non-null  object 
 10  Last Updated    10839 non-null  object 
 11  Current Ver     10831 non-null  object 
 12  Android Ver     10837 non-null  object 
dtypes: float64(2), int64(3), object(8)
memory usage: 1.2+ MB


In [186]:
playstore_df.sample(10)

App            Category  \
4213                         My Android Device S/W & H/W              FAMILY   
6554                                          Skater Boy                GAME   
7217                                           CE-STRONG              FAMILY   
2304                                             MyChart             MEDICAL   
4709                              Grand Stickman Cover V                GAME   
6673   Bleacher Report: sports news, scores, & highli...              SPORTS   
10256                         Barcelona FC Noticias Live  NEWS_AND_MAGAZINES   
1331                                   Yoga - Track Yoga  HEALTH_AND_FITNESS   
9393                                     Guess The Emoji                GAME   
8158                                           Strava.cz            SHOPPING   

       Rating  Reviews      Size   Installs  Type  Price Content Rating  \
4213      4.5      681   2800000      50000  Free    0.0       Everyone   
6554      4.3  1167143  12000000  100000000  Free    0.0       Everyone   
7217      NaN        0  16000000        100  Free    0.0       Everyone   
2304      4.2    19473         0    1000000  Free    0.0       Everyone   
4709      4.0     1744  37000000     100000  Free    0.0     Mature 17+   
6673      4.4   122280         0    5000000  Free    0.0   Everyone 10+   
10256     NaN        2   3600000        500  Free    0.0       Everyone   
1331      4.5     7976  44000000     500000  Free    0.0       Everyone   
9393      4.5    76627  23000000    5000000  Free    0.0       Everyone   
8158      4.3     2221         0     100000  Free    0.0       Everyone   

                 Genres        Last Updated         Current Ver  \
4213          Education   November 20, 2015                 1.1   
6554             Arcade  September 20, 2017             1.18.35   
7217          Education       June 17, 2016               1.0.4   
2304            Medical       July 24, 2018  Varies with device   
4709             Action        May 19, 2018               1.0.1   
6673             Sports       July 20, 2018  Varies with device   
10256  News & Magazines    February 4, 2018                 2.0   
1331   Health & Fitness        July 9, 2018                 7.0   
9393               Word        July 3, 2018               8.16g   
8158           Shopping       July 10, 2018  Varies with device   

              Android Ver  
4213           2.3 and up  
6554           4.1 and up  
7217           4.0 and up  
2304   Varies with device  
4709           4.1 and up  
6673   Varies with device  
10256        4.0.3 and up  
1331         4.0.3 and up  
9393           4.2 and up  
8158   Varies with device

In [187]:
playstore_df.isna().sum()

App                  0
Category             0
Rating            1473
Reviews              0
Size                 0
Installs             0
Type                 0
Price                0
Content Rating       0
Genres               0
Last Updated         0
Current Ver          8
Android Ver          2
dtype: int64

As we can see that in "Rating" there are 1473 null values. Hence we decided to replace them with the median of "Rating".

In [188]:
# Replacing the null values with the median of Rating:
def replace_nullval(nullval):
    """
    replace the rows with null values 
    with median values
    """
    return nullval.fillna(nullval.median())

playstore_df['Rating'] = playstore_df['Rating'].transform(replace_nullval)
playstore_df['Rating'] = playstore_df['Rating'].astype(float)

In [189]:
playstore_df.isna().sum()


App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       8
Android Ver       2
dtype: int64

In [190]:
playstore_df

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating  Reviews      Size  Installs  Type  Price Content Rating  \
0         4.1      159  19000000     10000  Free    0.0       Everyone   
1         3.9      967  14000000    500000  Free    0.0       Everyone   
2         4.7    87510   8700000   5000000  Free    0.0       Everyone   
3         4.5   215644  25000000  50000000  Free    0.0           Teen   
4         4.3      967   2800000    100000  Free    0.0       Everyone   
...       ...      ...       ...       ...   ...    ...            ...   
10836     4.5       38  53000000      5000  Free    0.0       Everyone   
10837     5.0        4   3600000       100  Free    0.0       Everyone   
10838     4.3        3   9500000      1000  Free    0.0       Everyone   
10839     4.5      114         0      1000  Free    0.0     Mature 17+   
10840     4.5   398307  19000000  10000000  Free    0.0       Everyone   

                          Genres      Last Updated         Current Ver  \
0                   Art & Design   January 7, 2018               1.0.0   
1      Art & Design;Pretend Play  January 15, 2018               2.0.0   
2                   Art & Design    August 1, 2018               1.2.4   
3                   Art & Design      June 8, 2018  Varies with device   
4        Art & Design;Creativity     June 20, 2018                 1.1   
...                          ...               ...                 ...   
10836                  Education     July 25, 2017                1.48   
10837                  Education      July 6, 2018                 1.0   
10838                    Medical  January 20, 2017                 1.0   
10839          Books & Reference  January 19, 2015  Varies with device   
10840                  Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
0            4.0.3 and up  
1            4.0.3 and up  
2            4.0.3 and up  
3              4.2 and up  
4              4.4 and up  
...                   ...  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device  

[10839 rows x 13 columns]

# 6. Cleaning the data of User Review Dataframe

In [191]:
#Eliminating the null value rows from the database

no_null_user_reviews_df = users_review_df[users_review_df['Sentiment'].isna()]
no_null_user_reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26863 entries, 2 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     26863 non-null  object 
 1   Translated_Review       0 non-null      object 
 2   Sentiment               0 non-null      object 
 3   Sentiment_Polarity      0 non-null      float64
 4   Sentiment_Subjectivity  0 non-null      float64
dtypes: float64(2), object(3)
memory usage: 1.2+ MB


In [192]:
no_null_user_reviews_df

App Translated_Review Sentiment  \
2            10 Best Foods for You               NaN       NaN   
7            10 Best Foods for You               NaN       NaN   
15           10 Best Foods for You               NaN       NaN   
102          10 Best Foods for You               NaN       NaN   
107          10 Best Foods for You               NaN       NaN   
...                            ...               ...       ...   
64290  Houzz Interior Design Ideas               NaN       NaN   
64291  Houzz Interior Design Ideas               NaN       NaN   
64292  Houzz Interior Design Ideas               NaN       NaN   
64293  Houzz Interior Design Ideas               NaN       NaN   
64294  Houzz Interior Design Ideas               NaN       NaN   

       Sentiment_Polarity  Sentiment_Subjectivity  
2                     NaN                     NaN  
7                     NaN                     NaN  
15                    NaN                     NaN  
102                   NaN                     NaN  
107                   NaN                     NaN  
...                   ...                     ...  
64290                 NaN                     NaN  
64291                 NaN                     NaN  
64292                 NaN                     NaN  
64293                 NaN                     NaN  
64294                 NaN                     NaN  

[26863 rows x 5 columns]

# 7. Preparation of the Data

In [193]:
# Checking for duplicate apps from the DataFrame:
duplicate_check = playstore_df['App'].duplicated().any()
duplicate_check

True

In [194]:
# Apps and count of each app
playstore_df['App'].value_counts()

ROBLOX                                               9
CBS Sports App - Scores, News, Stats & Watch Live    8
ESPN                                                 7
Duolingo: Learn Languages Free                       7
Candy Crush Saga                                     7
                                                    ..
U-Report                                             1
U of I Community Credit Union                        1
Waiting For U Launcher Theme                         1
Wi u Emulator                                        1
iHoroscope - 2018 Daily Horoscope & Astrology        1
Name: App, Length: 9658, dtype: int64

In [195]:
# Taking all the last rows of data from each app:
last_rows = playstore_df.groupby('App').tail(1).reset_index()
app_max_review = last_rows.loc[last_rows.groupby(['App'])['Reviews'].idxmax()]

In [196]:
#Getting the  Genres:
top_genres = app_max_review.Genres.value_counts().reset_index().rename(columns={'Genres':'Count','index':'Genres'})
top_genres

Genres  Count
0                                  Tools    828
1                          Entertainment    561
2                              Education    510
3                               Business    419
4                                Medical    396
..                                   ...    ...
113  Health & Fitness;Action & Adventure      1
114            Art & Design;Pretend Play      1
115    Travel & Local;Action & Adventure      1
116                     Trivia;Education      1
117         Books & Reference;Creativity      1

[118 rows x 2 columns]

In [197]:
# Reviewing The apps which are free to use based on reviews:
app_max_review[app_max_review['Price'] == 0]
app_max_review

index                                               App       Category  \
7703   8884              "i DT" Fútbol. Todos Somos Técnicos.         SPORTS   
7351   8532                     +Download 4 Instagram Twitter         SOCIAL   
267     324                        - Free Comics - Comic Apps         COMICS   
3360   4541                                                .R          TOOLS   
3455   4636                                            /u/app  COMMUNICATION   
...     ...                                               ...            ...   
5153   6334                           뽕티비 - 개인방송, 인터넷방송, BJ방송  VIDEO_PLAYERS   
3181   4362                                        💎 I'm rich      LIFESTYLE   
1774   2575     💘 WhatsLov: Smileys of love, stickers and GIF         SOCIAL   
6378   7559  📏 Smart Ruler ↔️ cm/inch measuring for homework!          TOOLS   
628     882  🔥 Football Wallpapers 4K | Full HD Backgrounds 😍  ENTERTAINMENT   

      Rating  Reviews      Size  Installs  Type   Price Content Rating  \
7703     4.3       27   3600000       500  Free    0.00       Everyone   
7351     4.5    40467  22000000   1000000  Free    0.00       Everyone   
267      3.5      115   9100000     10000  Free    0.00     Mature 17+   
3360     4.5      259    203000     10000  Free    0.00       Everyone   
3455     4.7      573  53000000     10000  Free    0.00     Mature 17+   
...      ...      ...       ...       ...   ...     ...            ...   
5153     4.3      414  59000000    100000  Free    0.00     Mature 17+   
3181     3.8      718  26000000     10000  Paid  399.99       Everyone   
1774     4.6    22098  18000000   1000000  Free    0.00       Everyone   
6378     4.0       19   3200000     10000  Free    0.00       Everyone   
628      4.7    11661   4000000   1000000  Free    0.00       Everyone   

                       Genres        Last Updated Current Ver   Android Ver  
7703                   Sports     October 7, 2017        0.22    4.1 and up  
7351                   Social      August 2, 2018        5.03    4.1 and up  
267                    Comics       July 13, 2018      5.0.12    5.0 and up  
3360                    Tools  September 16, 2014      1.1.06    1.5 and up  
3455            Communication        July 3, 2018       4.2.4    4.1 and up  
...                       ...                 ...         ...           ...  
5153  Video Players & Editors       July 18, 2018       4.0.7  4.0.3 and up  
3181                Lifestyle      March 11, 2018       1.0.0    4.4 and up  
1774                   Social       July 24, 2018       4.2.4  4.0.3 and up  
6378                    Tools    October 21, 2017         1.0    4.2 and up  
628             Entertainment       July 14, 2018     1.1.3.2  4.0.3 and up  

[9658 rows x 14 columns]

In [198]:
#Printing the dataframe which contains paid apps install counts:
genres_paidapp_count = playstore_df[playstore_df['Price']!= 0].groupby(['Genres'])[['Installs']].sum().rename(columns={'Installs':'Paid_app_installs'})
genres_paidapp_count

Paid_app_installs
Genres                                         
Action                                 17052250
Action;Action & Adventure                 20500
Adventure                               1278200
Adventure;Action & Adventure             150000
Adventure;Brain Games                    100000
...                                         ...
Strategy;Action & Adventure              100000
Tools                                   1727441
Travel & Local                           183060
Video Players & Editors                   71000
Weather                                  812000

[72 rows x 1 columns]

In [199]:
#Printing the  dataframe which contains mean Rating
genres_ratings_df = app_max_review.groupby(['Genres'])[['Rating']].mean()
genres_ratings_df

Rating
Genres                                         
Action                                 4.246154
Action;Action & Adventure              4.316667
Adventure                              4.182192
Adventure;Action & Adventure           4.300000
Adventure;Brain Games                  4.600000
...                                         ...
Trivia;Education                       4.300000
Video Players & Editors                4.066871
Video Players & Editors;Music & Video  4.000000
Weather                                4.248101
Word                                   4.317391

[118 rows x 1 columns]

In [200]:
#Preparing dataframe which contains free app install counts
genres_freeapp_install = app_max_review[app_max_review['Price'] == 0].groupby(['Genres'])[['Installs']].sum()


In [201]:
#Mergering all the data previous dataframes
top_genres_install_df = pd.merge(top_genres, genres_freeapp_install, on='Genres')
top_genres_apps_installs_df = pd.merge(top_genres_install_df, genres_paidapp_count, on='Genres')
top_genres_apps_installs_ratings_df = pd.merge(top_genres_apps_installs_df, genres_ratings_df, on='Genres')

#Getting top 50 data frames based on the Genres
top_50_genres_df = top_genres_apps_installs_ratings_df.head(50)
top_50_genres_df

Genres  Count     Installs  Paid_app_installs    Rating
0                       Tools    828   8091044474            1727441  4.073913
1               Entertainment    561   3014472513             242171  4.118182
2                   Education    510    262803901             106247  4.292353
3                    Business    419    691952090             412775  4.172792
4                     Medical    396     38642344            1020033  4.204040
5             Personalization    376   1529235988            5258794  4.325532
6                Productivity    374   5786679314            1412055  4.206150
7                   Lifestyle    368    492644429            1179110  4.131793
8                     Finance    345    455163132             185602  4.138551
9                      Sports    331   1411230683            1243815  4.238369
10              Communication    316  11037916201            1360050  4.154747
11                     Action    299   3465986940           17052250  4.246154
12           Health & Fitness    287   1133548402             574110  4.250871
13                Photography    281   4656268915            3978740  4.165125
14           News & Magazines    254   2369212260               5500  4.156693
15                     Social    239   5487861902               6000  4.255230
16          Books & Reference    222   1665946260              23316  4.334234
17             Travel & Local    218   2894604086             183060  4.103211
18                   Shopping    202   1400338685              10100  4.238119
19                 Simulation    193    629362620             235200  4.145596
20                     Arcade    185   3758691940            1437215  4.275135
21                     Dating    170    140914757              21350  4.050000
22                     Casual    164   3042798570              32170  4.105488
23    Video Players & Editors    163   3916931720              71000  4.066871
24          Maps & Navigation    131    503160790             121100  4.062595
25                     Puzzle    119    830286191            2190620  4.358824
26               Food & Drink    112    211738751              60000  4.191964
27               Role Playing    105    329148570            2092100  4.272381
28                   Strategy     94    912192105            1995500  4.217021
29                     Racing     91   1400136820             110500  4.158242
30            Auto & Vehicles     85     53080061              50150  4.205882
31           Libraries & Demo     84     52995810                100  4.207143
32                    Weather     79    360288520             812000  4.248101
33                  Adventure     73    295417120            1278200  4.182192
34                     Events     64     15973160                  1  4.395313
35               Art & Design     57    112512100              16000  4.356140
36                  Parenting     46     20591010              50100  4.334783
37                       Card     46    142618500             207600  4.073913
38                      Board     42    161813110             314100  4.280952
39                     Casino     39    130260600                100  4.287179
40      Educational;Education     38     60800010              11050  4.250000
41                Educational     37     13569100                206  3.929730
42        Education;Education     36    142795510              18050  4.294444
43        Casual;Pretend Play     25    146100000             261000  4.152000
44                      Music     19    170020500             100000  4.173684
45         Puzzle;Brain Games     17    139210000            1100000  4.370588
46     Education;Pretend Play     17      9000000            1188000  4.358824
47  Racing;Action & Adventure     16    132250000              10000  4.287500
48  Arcade;Action & Adventure     14     35100001           20110000  4.335714
49          Board;Brain Games     14      2850000              32100  4.3